src/Workspaces/Core/Portable/Shared/Utilities/ProducerConsumer.cs

```diff
@@ -42,21 +42,21 @@ internal static class ProducerConsumer<TItem>
     /// </summary>
     public static Task RunAsync<TArgs>(
         ProducerConsumerOptions options,
-        Func<Action<TItem>, TArgs, Task> produceItems,
-        Func<ImmutableArray<TItem>, TArgs, Task> consumeItems,
+        Func<Action<TItem>, TArgs, CancellationToken, Task> produceItems,
```

Comment:
>produceItems [](http://example.com/codeflow?start=60&length=12)

naming nit: maybe produceItem instead?

Reply:
I like produceItems. It's the callback responsible for producing *all* the items. It will just do that by producing them and passing them to callback it itself is provided.